[BUG]: "default" export missing from package.json

### What happened?

When trying to compile my github action with ncc, an error stated that this plugin (and two others) are missing the "default" export inside package.json -> exports['.']. I believe this is because the scripts/build.mjs has not gotten the latest changes from octokit/core as shown here: https://github.com/octokit/core.js/blob/main/scripts/build.mjs#L63



### Versions

- Node v21.4.0
- Typescript 5.4.5 (also tried 4.7.4)
- @vercel/ncc v0.38.1
- @octokit/plugin-paginate-graphql v5.2.0
- @octokit/plugin-retry v7.1.0
- @octokit/plugin-throttling v9.2.0

### Relevant log output

```shell
> ncc build ./src/action.ts -o dist -d
 
ncc: Version 0.38.1
ncc: Compiling file index.js into CJS
ncc: Using typescript@5.4.5 (local user-provided)
Error: Module not found: Error: Package path . is not exported from package /Users/kyle/git/actions/octokit_project/node_modules/@octokit/plugin-paginate-graphql (see exports field in /Users/kyle/git/actions/octokit_project/node_modules/@octokit/plugin-paginate-graphql/package.json)
Did you mean './@octokit/plugin-paginate-graphql'?
Requests that should resolve in the current directory need to start with './'.
Requests that start with a name are treated as module requests and resolve within module directories (node_modules).
If changing the source code is not an option there is also a resolve options called 'preferRelative' which tries to resolve these kind of requests in the current directory too.
Module not found: Error: Package path . is not exported from package /Users/kyle/git/actions/octokit_project/node_modules/@octokit/plugin-throttling (see exports field in /Users/kyle/git/actions/octokit_project/node_modules/@octokit/plugin-throttling/package.json)
Did you mean './@octokit/plugin-throttling'?
Requests that should resolve in the current directory need to start with './'.
Requests that start with a name are treated as module requests and resolve within module directories (node_modules).
If changing the source code is not an option there is also a resolve options called 'preferRelative' which tries to resolve these kind of requests in the current directory too.
Module not found: Error: Package path . is not exported from package /Users/kyle/git/actions/octokit_project/node_modules/@octokit/plugin-retry (see exports field in /Users/kyle/git/actions/octokit_project/node_modules/@octokit/plugin-retry/package.json)
Did you mean './@octokit/plugin-retry'?
Requests that should resolve in the current directory need to start with './'.
Requests that start with a name are treated as module requests and resolve within module directories (node_modules).
If changing the source code is not an option there is also a resolve options called 'preferRelative' which tries to resolve these kind of requests in the current directory too.
    at /Users/kyle/git/actions/octokit_project/node_modules/@vercel/ncc/dist/ncc/index.js.cache.js:38:1896272
    at /Users/kyle/git/actions/octokit_project/node_modules/@vercel/ncc/dist/ncc/index.js.cache.js:38:396262
    at _done (eval at create (/Users/kyle/git/actions/octokit_project/node_modules/@vercel/ncc/dist/ncc/index.js.cache.js:21:75523), <anonymous>:9:1)
    at eval (eval at create (/Users/kyle/git/actions/octokit_project/node_modules/@vercel/ncc/dist/ncc/index.js.cache.js:21:75523), <anonymous>:34:22)
```

### Related Bugs
https://github.com/octokit/plugin-retry.js/issues/541
https://github.com/octokit/plugin-paginate-graphql.js/issues/194

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/README.md
+++ b/README.md
@@ -42,6 +42,12 @@ import { throttling } from "@octokit/plugin-throttling";
 </tbody>
 </table>
 
+> [!IMPORTANT]
+> As we use [conditional exports](https://nodejs.org/api/packages.html#conditional-exports), you will need to adapt your `tsconfig.json` by setting `"moduleResolution": "node16", "module": "node16"`.
+>
+> See the TypeScript docs on [package.json "exports"](https://www.typescriptlang.org/docs/handbook/modules/reference.html#packagejson-exports).<br>
+> See this [helpful guide on transitioning to ESM](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c) from [@sindresorhus](https://github.com/sindresorhus)
+
 The code below creates a "Hello, world!" issue on every repository in a given organization. Without the throttling plugin it would send many requests in parallel and would hit rate limits very quickly. But the `@octokit/plugin-throttling` slows down your requests according to the official guidelines, so you don't get blocked before your quota is exhausted.
 
 The `throttle.onSecondaryRateLimit` and `throttle.onRateLimit` options are required. Return `true` to automatically retry the request after `retryAfter` seconds.

--- a/scripts/build.mjs
+++ b/scripts/build.mjs
@@ -61,15 +61,13 @@ async function main() {
       {
         ...pkg,
         files: ["dist-*/**", "bin/**"],
-        // Tooling currently are having issues with the "exports" field, ex: TypeScript, eslint
-        // We add a `main` and `types` field to the package.json for the time being
-        // See https://github.com/octokit/core.js/pulls/662
-        main: "dist-bundle/index.js",
         types: "dist-types/index.d.ts",
         exports: {
           ".": {
             types: "./dist-types/index.d.ts",
             import: "./dist-bundle/index.js",
+            // Tooling currently are having issues with the "exports" field when there is no "default", ex: TypeScript, eslint, ncc
+            // See https://github.com/octokit/core.js/issues/667#issuecomment-2037592361
             default: "./dist-bundle/index.js",
           },
         },

--- a/scripts/build.mjs
+++ b/scripts/build.mjs
@@ -70,6 +70,7 @@ async function main() {
           ".": {
             types: "./dist-types/index.d.ts",
             import: "./dist-bundle/index.js",
+            default: "./dist-bundle/index.js",
           },
         },
         sideEffects: false,

--- a/scripts/build.mjs
+++ b/scripts/build.mjs
@@ -68,6 +68,7 @@ async function main() {
             import: "./dist-bundle/index.js",
             // Tooling currently are having issues with the "exports" field when there is no "default", ex: TypeScript, eslint, ncc
             // See https://github.com/octokit/core.js/issues/667#issuecomment-2037592361
+            // See https://github.com/octokit/plugin-throttling.js/issues/694
             default: "./dist-bundle/index.js",
           },
         },

